// @salutejs/plasma-tokens/themes/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
var darkMetro_1 = require("./darkMetro");
Object.defineProperty(exports, "darkMetro", { enumerable: true, get: function () { return darkMetro_1.darkMetro; } });
var lightMetro_1 = require("./lightMetro");
Object.defineProperty(exports, "lightMetro", { enumerable: true, get: function () { return lightMetro_1.lightMetro; } });
var darkSbermarket_1 = require("./darkSbermarket");
Object.defineProperty(exports, "darkSbermarket", { enumerable: true, get: function () { return darkSbermarket_1.darkSbermarket; } });
var lightSbermarket_1 = require("./lightSbermarket");
Object.defineProperty(exports, "lightSbermarket", { enumerable: true, get: function () { return lightSbermarket_1.lightSbermarket; } });
var darkSberprime_1 = require("./darkSberprime");
Object.defineProperty(exports, "darkSberprime", { enumerable: true, get: function () { return darkSberprime_1.darkSberprime; } });
var lightSberprime_1 = require("./lightSberprime");
Object.defineProperty(exports, "lightSberprime", { enumerable: true, get: function () { return lightSberprime_1.lightSberprime; } });
var darkSelgros_1 = require("./darkSelgros");
Object.defineProperty(exports, "darkSelgros", { enumerable: true, get: function () { return darkSelgros_1.darkSelgros; } });
var lightSelgros_1 = require("./lightSelgros");
Object.defineProperty(exports, "lightSelgros", { enumerable: true, get: function () { return lightSelgros_1.lightSelgros; } });
var darkSmbusiness_1 = require("./darkSmbusiness");
Object.defineProperty(exports, "darkSmbusiness", { enumerable: true, get: function () { return darkSmbusiness_1.darkSmbusiness; } });
var lightSmbusiness_1 = require("./lightSmbusiness");
Object.defineProperty(exports, "lightSmbusiness", { enumerable: true, get: function () { return lightSmbusiness_1.lightSmbusiness; } });
var darkSber_1 = require("./darkSber");
Object.defineProperty(exports, "darkSber", { enumerable: true, get: function () { return darkSber_1.darkSber; } });
var darkEva_1 = require("./darkEva");
Object.defineProperty(exports, "darkEva", { enumerable: true, get: function () { return darkEva_1.darkEva; } });
var darkJoy_1 = require("./darkJoy");
Object.defineProperty(exports, "darkJoy", { enumerable: true, get: function () { return darkJoy_1.darkJoy; } });
var darkBrand_1 = require("./darkBrand");
Object.defineProperty(exports, "darkBrand", { enumerable: true, get: function () { return darkBrand_1.darkBrand; } });
var darkB2E_1 = require("./darkB2E");
Object.defineProperty(exports, "darkB2E", { enumerable: true, get: function () { return darkB2E_1.darkB2E; } });
var lightSber_1 = require("./lightSber");
Object.defineProperty(exports, "lightSber", { enumerable: true, get: function () { return lightSber_1.lightSber; } });
var lightEva_1 = require("./lightEva");
Object.defineProperty(exports, "lightEva", { enumerable: true, get: function () { return lightEva_1.lightEva; } });
var lightJoy_1 = require("./lightJoy");
Object.defineProperty(exports, "lightJoy", { enumerable: true, get: function () { return lightJoy_1.lightJoy; } });
var lightBrand_1 = require("./lightBrand");
Object.defineProperty(exports, "lightBrand", { enumerable: true, get: function () { return lightBrand_1.lightBrand; } });
var lightB2E_1 = require("./lightB2E");
Object.defineProperty(exports, "lightB2E", { enumerable: true, get: function () { return lightB2E_1.lightB2E; } });
