// @salutejs/plasma-core/es/components/Popup/Popup.js
var _excluded = ["disclosure", "children", "isOpen", "trigger", "placement", "onToggle"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { memo, useRef, useCallback, useEffect } from 'react';
import styled, { css } from 'styled-components';
import { useForkRef } from '../../hooks';
var StyledRoot = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma__sc-w3yec9-0"
})(["position:relative;box-sizing:border-box;display:inline-flex;"]);
var StyledPopup = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma__sc-w3yec9-1"
})(["position:absolute;z-index:1;padding:var(--plasma-popup-padding);margin:var(--plasma-popup-margin);width:var(--plasma-popup-width);", ""], function (_ref) {
  var placement = _ref.placement;

  switch (placement) {
    case 'left':
      return css(["top:0;right:100%;"]);

    case 'right':
      return css(["top:0;left:100%;"]);

    case 'bottom':
    default:
      return css(["top:100%;left:0;"]);
  }
});
/**
 * Всплывающее окно с возможностью позиционирования
 * и вызова по клику либо ховеру.
 */

export var Popup = /*#__PURE__*/memo( /*#__PURE__*/React.forwardRef(function (_ref2, outerRootRef) {
  var disclosure = _ref2.disclosure,
      children = _ref2.children,
      isOpen = _ref2.isOpen,
      trigger = _ref2.trigger,
      placement = _ref2.placement,
      onToggle = _ref2.onToggle,
      rest = _objectWithoutProperties(_ref2, _excluded);

  var rootRef = useRef(null);
  var popupRef = useRef(null);
  var handleRef = useForkRef(rootRef, outerRootRef);
  var onDocumentClick = useCallback(function (event) {
    var _rootRef$current;

    var targetIsRoot = event.target === rootRef.current;
    var rootHasTarget = (_rootRef$current = rootRef.current) === null || _rootRef$current === void 0 ? void 0 : _rootRef$current.contains(event.target);

    if (!targetIsRoot && !rootHasTarget) {
      onToggle === null || onToggle === void 0 ? void 0 : onToggle(false, event);
    }
  }, [onToggle]);
  var onClick = useCallback(function (event) {
    if (trigger === 'click') {
      var _popupRef$current;

      var targetIsPopup = event.target === popupRef.current;
      var rootHasTarget = (_popupRef$current = popupRef.current) === null || _popupRef$current === void 0 ? void 0 : _popupRef$current.contains(event.target);

      if (!targetIsPopup && !rootHasTarget) {
        onToggle === null || onToggle === void 0 ? void 0 : onToggle(!isOpen, event);
      }
    }
  }, [trigger, isOpen, onToggle]);
  var onMouseEnter = useCallback(function (event) {
    if (trigger === 'hover') {
      onToggle === null || onToggle === void 0 ? void 0 : onToggle(true, event);
    }
  }, [trigger, onToggle]);
  var onMouseLeave = useCallback(function (event) {
    if (trigger === 'hover') {
      onToggle === null || onToggle === void 0 ? void 0 : onToggle(false, event);
    }
  }, [trigger, onToggle]);
  var onFocus = useCallback(function (event) {
    if (trigger === 'hover') {
      onToggle === null || onToggle === void 0 ? void 0 : onToggle(true, event);
    }
  }, [trigger, onToggle]);
  var onBlur = useCallback(function (event) {
    if (trigger === 'hover') {
      onToggle === null || onToggle === void 0 ? void 0 : onToggle(false, event);
    }
  }, [trigger, onToggle]);
  useEffect(function () {
    document.addEventListener('click', onDocumentClick);
    return function () {
      return document.removeEventListener('click', onDocumentClick);
    };
  }, []);
  return /*#__PURE__*/React.createElement(StyledRoot, _extends({
    ref: handleRef,
    onClick: onClick,
    onMouseEnter: onMouseEnter,
    onMouseLeave: onMouseLeave,
    onFocus: onFocus,
    onBlur: onBlur
  }, rest), disclosure, children && /*#__PURE__*/React.createElement(StyledPopup, {
    ref: popupRef,
    placement: placement,
    style: {
      display: isOpen ? 'block' : 'none'
    }
  }, children));
}));
