// @salutejs/plasma-ui/components/TextField/TextField.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TextField = void 0;

var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");

var _Field = /*#__PURE__*/require("../Field");

var _utils = /*#__PURE__*/require("../../utils");

var _excluded = ["id", "size", "disabled", "status", "label", "placeholder", "contentLeft", "contentRight", "helperText", "style", "className", "onChange", "onSearch"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var StyledInput = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.Input).attrs(_utils.spatnavClassNameAttrs).withConfig({
  componentId: "plasma-ui__sc-1o5gwii-0"
})(["", ";border-radius:1rem;&::placeholder{opacity:0;}&:placeholder-shown + ", "{color:", ";}"], _Field.applyInputStyles, _plasmaCore.FieldPlaceholder, function (_ref) {
  var status = _ref.status;
  return status ? null : _plasmaCore.secondary;
});
/**
 * Поле ввода текста.
 */
// eslint-disable-next-line prefer-arrow-callback

var TextField = /*#__PURE__*/(0, _react.forwardRef)(function TextField(_ref2, ref) {
  var id = _ref2.id,
      _ref2$size = _ref2.size,
      size = _ref2$size === void 0 ? 'l' : _ref2$size,
      disabled = _ref2.disabled,
      status = _ref2.status,
      label = _ref2.label,
      placeholder = _ref2.placeholder,
      contentLeft = _ref2.contentLeft,
      contentRight = _ref2.contentRight,
      helperText = _ref2.helperText,
      style = _ref2.style,
      className = _ref2.className,
      onChange = _ref2.onChange,
      onSearch = _ref2.onSearch,
      rest = _objectWithoutProperties(_ref2, _excluded);

  var placeLabel = label || placeholder;
  var handleChange = (0, _react.useCallback)(function (event) {
    var _event$target = event.target,
        maxLength = _event$target.maxLength,
        value = _event$target.value;

    if (!onChange || maxLength !== -1 && value.length > maxLength) {
      return;
    }

    onChange(event);
  }, [onChange]);
  var handleSearch = (0, _react.useCallback)(function (event) {
    if (event.keyCode === 13 && onSearch) {
      onSearch(event.target.value, event);
    }
  }, [onSearch]);
  return /*#__PURE__*/_react["default"].createElement(_plasmaCore.FieldRoot, {
    $size: size,
    $disabled: disabled,
    $isContentLeft: Boolean(contentLeft),
    $isContentRight: Boolean(contentRight),
    $isHelper: Boolean(helperText),
    status: status,
    style: style,
    className: className
  }, contentLeft && /*#__PURE__*/_react["default"].createElement(_plasmaCore.FieldContent, {
    pos: "left"
  }, contentLeft), /*#__PURE__*/_react["default"].createElement(StyledInput, _extends({
    $size: size,
    ref: ref,
    id: id,
    placeholder: placeLabel,
    disabled: disabled,
    status: status,
    "aria-describedby": id ? "".concat(id, "-helpertext") : undefined,
    onChange: handleChange,
    onKeyUp: onSearch && handleSearch
  }, rest)), placeLabel && size === 'l' && /*#__PURE__*/_react["default"].createElement(_plasmaCore.FieldPlaceholder, {
    htmlFor: id
  }, placeLabel), contentRight && /*#__PURE__*/_react["default"].createElement(_plasmaCore.FieldContent, {
    pos: "right"
  }, contentRight), helperText && /*#__PURE__*/_react["default"].createElement(_Field.FieldHelper, {
    id: id ? "".concat(id, "-helpertext") : undefined
  }, helperText));
});
exports.TextField = TextField;
